src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java

```diff
@@ -26,82 +21,115 @@
+    private final JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
 
     private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-    private final PrintStream originalOut = System.out;
 
     @BeforeEach
     public void setUp() {
-        System.setOut(new PrintStream(byteArrayOutputStream));
+        environmentVariables.put("PATH", pathEnvVarInput);
+        Mockito.when(taskListener.getLogger()).thenReturn(new PrintStream(byteArrayOutputStream));
     }
 
-    @AfterEach
-    public void tearDown() {
-        System.setOut(originalOut);
+    @Test
+    public void testSetUpForExecutionEmptySpaceStringInput() {
+        jenkinsIntLogger.setLogLevel(LogLevel.DEBUG);
+        this.createAndTestDetectSetupResponse(" ", " ", javaPathSystem);
+        this.debugLogLevelValidation(javaPathSystem, " ");
+        this.invalidJavaExecValidation();
     }
 
-    private static Stream<Arguments> setUpForExecutionInput() {
-        return Stream.of(
-            Arguments.of(null, true, null, testPathEnv, LogLevel.DEBUG),
-            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.WARN),
-            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.INFO),
-            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.DEBUG),
-            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.TRACE),
-            Arguments.of("", false, "", "", LogLevel.DEBUG),
-            Arguments.of(" ", false, " ", " ", LogLevel.DEBUG),
-            Arguments.of("", false, "", "", null)
-        );
+    @Test
+    public void testSetUpForExecutionEmptyStringInput() {
+        jenkinsIntLogger.setLogLevel(LogLevel.DEBUG);
+        this.createAndTestDetectSetupResponse("", "", javaBinPath);
+        this.debugLogLevelValidation(javaBinPath, "");
+        this.invalidJavaExecValidation();
     }
 
-    @ParameterizedTest
-    @MethodSource({ "setUpForExecutionInput" })
-    public void testSetUpForExecution(String javaHomeInput, Boolean isValidJavaHome, String detectJarPathInput, String pathEnvInput, LogLevel logLevel) {
-        Map<String, String> environmentVariables = new HashMap<>();
-        environmentVariables.put("PATH", pathEnvInput);
-
-        TaskListener taskListener = Mockito.mock(TaskListener.class);
-        Mockito.when(taskListener.getLogger()).thenReturn(new PrintStream(byteArrayOutputStream));
-        JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
-
-        if (logLevel != null) {
-            jenkinsIntLogger.setLogLevel(logLevel);
-        }
-
-        try {
-            // Perform same actions as calculateJavaExecutablePath().
-            String adjustedJavaHome = javaHomeInput == null ? "java" : new File(javaHomeInput + "/bin/java").getCanonicalPath();
+    @Test
+    public void testSetUpForExecutionNullInput() {
```

Comment:
At first glance, this test and the two tests preceding it are the same test, but with a different java home / jar path. However, looking further into it, you're also testing different Java paths and asserting different results. Unless there's a compelling reason not to, it seems likely that we should combine these three tests into a parameterized test that tests " ", "", and null across the same other lines of test code.

Reply:
The 3rd test (testSetUpForExecutionNullInput) calls this.**valid**JavaExecValidation(), the other 2 call this.**invalid**JavaExecValidation(). I didn't think the added complexity to handle both cases OR a parameterized test for 2 tests was worth it here.

Reply:
Ah, likely a good call. I missed the distinction in validation.